pkg/providerconfig/types.go

```diff
@@ -147,3 +155,22 @@ func NewConfigVarResolver(ctx context.Context, client ctrlruntimeclient.Client)
 		client: client,
 	}
 }
+
+func DefaultOperatingSystemSpec(osys providerconfigtypes.OperatingSystem, operatingSystemSpec runtime.RawExtension) (runtime.RawExtension, error) {
+	switch osys {
+	case providerconfigtypes.OperatingSystemAmazonLinux2:
+		return amzn2.DefaultConfig(operatingSystemSpec), nil
+	case providerconfigtypes.OperatingSystemCentOS:
+		return centos.DefaultConfig(operatingSystemSpec), nil
+	case providerconfigtypes.OperatingSystemFlatcar:
+		return flatcar.DefaultConfig(operatingSystemSpec), nil
+	case providerconfigtypes.OperatingSystemRHEL:
```

Comment:
This actually won't work for RHEL, the reason behind that is, the operating system spec holds all the needed information to register the os such as the username, password and offline token. Without those it won't be possible to get a running reel machine. I would suggest to throw an error when reel is used

Reply:
Yeah, good point

Reply:
done!

Reply:
>Without those it won't be possible to get a running rhel machine.
Turns out, on some clouds that's not the case (on AWS you could perfectly well have RHEL machines without registration). So the logic is restored to the original one.